java-checks/src/main/java/org/sonar/java/checks/AbstractClassNoFieldShouldBeInterfaceCheck.java

```diff
@@ -74,10 +74,14 @@ private static boolean classIsAbstract(ClassTree tree) {
 
   private static boolean classHasNoFieldAndProtectedMethod(ClassTree tree) {
     for (Tree member : tree.members()) {
-      if (member.is(Tree.Kind.VARIABLE) || (member.is(Tree.Kind.METHOD) && ModifiersUtils.hasModifier(((MethodTree) member).modifiers(), Modifier.PROTECTED))) {
+      if (member.is(Tree.Kind.VARIABLE) || (member.is(Tree.Kind.METHOD) && isProtectedOrOverriding((MethodTreeImpl) member))) {
         return false;
       }
     }
     return true;
   }
+
+  private static boolean isProtectedOrOverriding(MethodTreeImpl member) {
+    return ModifiersUtils.hasModifier(member.modifiers(), Modifier.PROTECTED) || !Boolean.FALSE.equals(member.isOverriding());
```

Comment:
I guess that this also deserves a fix : https://github.com/SonarSource/sonar-java/blob/master/java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java#L269-269